### PR TITLE
Icons: compression fixes & larger dimension support

### PR DESCRIPTION
**Pull request by hedger**
# What's new

- gui: icon API improvements: getters for single frames
- gui: increased max supported dimensions for icons to 65535x65535 (from 255x255)
- gui, compression: unified heatshrink usage, added extra validations for icon decompression results 
- gui, compression: added support for decoding icons larger than decomp buffer size (1k), using dynamic allocations if needed
- unit_tests: reworked to run tests as separate modules from SD card
- fbt: added module filtering for firmware dist package
- loader: api: added md5_calc, bt_keys_storage, pulse_glue, varint

# Verification 

- Everything should work as before
- You can create code that uses large images

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: lib/toolbox/compress.c [hunk 1]
```diff
@@ -149,22 +167,19 @@ bool compress_encode(
...
     if(finish_res < 0) {
         encode_failed = true;
     } else {
         do {
             poll_res = heatshrink_encoder_poll(
-                compress->encoder,
-                &data_out[res_buff_size],
-                data_out_size - 4 - res_buff_size,
-                &poll_size);
+                encoder, &data_out[res_buff_size], data_out_size - 4 - res_buff_size, &poll_size);
```
Comment (DrZlo13) [inline lib/toolbox/compress.c]: `4`
magic number :(

File: applications/services/gui/icon.c [hunk 2]
```diff
@@ -16,5 +19,19 @@ uint8_t icon_get_height(const Icon* instance) {
...
-    return instance->frames[0];
+    return icon_get_frame_data(instance, 0);
+}
+
+uint32_t icon_get_frame_count(const Icon* instance) {
+    return instance->frame_count;
+}
+
+const uint8_t* icon_get_frame_data(const Icon* instance, uint32_t frame) {
+    furi_assert(frame < instance->frame_count);
```
Comment (skotopes) [inline applications/services/gui/icon.c]: furi_check

File: applications/services/gui/icon.h [hunk 3]
```diff
@@ -19,23 +20,48 @@ typedef struct Icon Icon;
...
+ * @return     frame count
+ */
+uint32_t icon_get_frame_count(const Icon* instance);
+
+/** Get Icon XBM bitmap data
+ *
+ * @param[in]  instance  pointer to Icon data
+ * @param[in]  frame     frame index
+ *
+ * @return     pointer to XBM bitmap data
```
Comment (skotopes) [inline applications/services/gui/icon.h]: Not XBM, flipper compressed XBM

File: applications/services/gui/icon.h [hunk 4]
```diff
@@ -19,23 +20,48 @@ typedef struct Icon Icon;
...
+uint16_t icon_get_height(const Icon* instance);
 
 /** Get Icon XBM bitmap data
  *
  * @param[in]  instance  pointer to Icon data
  *
  * @return     pointer to XBM bitmap data
  */
-const uint8_t* icon_get_data(const Icon* instance);
+FURI_DEPRECATED const uint8_t* icon_get_data(const Icon* instance);
```
Comment (skotopes) [inline applications/services/gui/icon.h:39]: Drop maybe? How many apps are actually using it?

File: lib/toolbox/compress.c [hunk 5]
```diff
@@ -35,42 +45,59 @@ CompressIcon* compress_icon_alloc(void) {
         COMPRESS_LOOKAHEAD_BUFF_SIZE_LOG);
     heatshrink_decoder_reset(instance->decoder);
     memset(instance->decoded_buff, 0, sizeof(instance->decoded_buff));
+    instance->large_decoded_buff = NULL;
 
     return instance;
 }
 
+static void compress_icon_release_large_buffer(CompressIcon* instance) {
+    furi_assert(instance);
```
Comment (skotopes) [inline lib/toolbox/compress.c]: furi_check in all public APIs

File: lib/toolbox/compress.c [hunk 6]
```diff
@@ -35,42 +45,59 @@ CompressIcon* compress_icon_alloc(void) {
         COMPRESS_LOOKAHEAD_BUFF_SIZE_LOG);
     heatshrink_decoder_reset(instance->decoder);
     memset(instance->decoded_buff, 0, sizeof(instance->decoded_buff));
+    instance->large_decoded_buff = NULL;
```
Comment (skotopes) [inline lib/toolbox/compress.c]: Pass advised buffer size and alloc buffer

File: lib/toolbox/compress.c [hunk 7]
```diff
@@ -35,42 +45,59 @@ CompressIcon* compress_icon_alloc(void) {
...
+    size_t size_hint) {
+    furi_assert(instance);
+    furi_assert(icon_data);
+    furi_assert(decoded_buff);
+
+    uint8_t* decoded_data_output_ptr = NULL;
+    size_t decoded_data_output_size = 0;
+    if(size_hint > COMPRESS_ICON_DECODED_BUFF_SIZE) {
+        compress_icon_release_large_buffer(instance);
+        instance->large_decoded_buff = malloc(size_hint);
```
Comment (skotopes) [inline lib/toolbox/compress.c]: realloc if needed

**Review by skotopes — CHANGES_REQUESTED** [review]

File: applications/services/gui/icon.h (hunk 4 above)
Comment (WillyJL) [reply applications/services/gui/icon.h:39]: i see its used in loader_applications, for setting js icon when not a fap. also i see Quac! app uses it for similar purpose

Comment (skotopes) [reply applications/services/gui/icon.h:39]: @Willy-JL We'll keep it for some time.

**Review by skotopes — APPROVED** [review]

**Pull request merged**